manifests/init.pp

```diff
@@ -126,5 +132,33 @@
       group  => $puppet_group,
       mode   => '0750',
     }
+    if $manage_fact_watcher_dependencies {
+      $ensure = if $fact_watcher_service {
+        'installed'
+      } else {
+        'absent'
+      }
+      package { 'ruby-inotify':
+        ensure   => 'installed',
+        provider => 'puppet_gem',
+        before   => Systemd::Unit_file['fact_watcher.service'],
+      }
+    }
+    systemd::manage_unit { 'fact_watcher.service':
+      enable        => $fact_watcher_service,
+      active        => $fact_watcher_service,
+      unit_entry    => {
+        'Description' => 'Publish facts to Foreman',
+      },
+      service_entry => {
+        'Type'        => 'simple',
+        'Environment' => "PATH=/opt/puppetlabs/puppet/bin:${facts['path']}",
+        'User'        => $puppet_user,
+        'ExecStart'   => "${puppet_etcdir}/node.rb --watch-facts --push-facts-parallel",
```

Comment:
If you want to enforce the AIO Ruby (and I really dislike it), perhaps this is better?
```suggestion
        'User'        => $puppet_user,
        'ExecStart'   => "/opt/puppetlabs/puppet/bin/ruby ${puppet_etcdir}/node.rb --watch-facts --push-facts-parallel",
```